### PR TITLE
Feature apollo client upgrade

### PR DESCRIPTION
**Pull request by puncsky**

### CODE REVIEW

File: src/shared/article/hooks/useArticles.ts [hunk 1]
```diff
@@ -0,0 +1,18 @@
+import { useQuery, ApolloError } from "@apollo/client";
+import { articles } from "@/shared/article/data/__generated__/articles";
+import { getArticles } from "@/shared/article/data/queries";
+
+export const useArticles = (
+  variables: { id: string } | null
+): {
+  loading: boolean;
+  error: ApolloError | undefined;
```
Comment (puncsky) [inline src/shared/article/hooks/useArticles.ts:9]: let's prefer

```
error?: ApolloError
```

File: src/shared/contact/delete-note-container.tsx [hunk 2]
```diff
@@ -76,7 +77,7 @@ export const DeleteNotePopover = withRouter(
               <Flex justifyContent="flex-start">
                 <Mutation mutation={DELETE_NOTE}>
                   {(
-                    deleteContact: MutationFn<{ id: string }>,
+                    deleteContact: MutationFunction<{ id: string }>,
```
Comment (puncsky) [inline src/shared/contact/delete-note-container.tsx:80]: need to migrate to `useMutation`

Comment (dzcpy) [reply src/shared/contact/delete-note-container.tsx:80]: Sure, I just temperately migrated these places for making it pass the test after upgrading to apollo client 3. Will fix them one by one later.

File: src/shared/article/data/queries.ts [hunk 3]
```diff
@@ -0,0 +1,10 @@
+import gql from "graphql-tag";
+
+export const getArticles = gql`
+  query articles($id: String!) {
```
[the remark this answers is not in the document]
Comment (dzcpy) [reply src/shared/article/data/queries.ts]: Fixed

**Pull request merged**